### DISCUSSION TITLE
URL appends previous pathname when current page has same pathname with dynamic routing.

### DISCUSSION THREAD

**Discussion by chen-chens** [vercel/next.js#46237] [Help] [answered]
## Environment Information
Next.js : `v12.3.1`

## Describe the Issue : 

I create a Navigation component in `_app.ts` with `Link` to change url.
Here is my dynamic folder: 
```
/products/categories/[productType]
```

- It works fine when url is not the same pathname.
`/products/details/123` to `/products/categories/1` : Output as expected `/products/categories/1`

- However, when the current page is on the same pathname:
`/products/categories/1` to another page `/products/categories/2`:
The URL will be **`/products/categories/products/categories/2`** instead of expected `/products/categories/2`.

Here is the `console.log()`
```
asPath: "/products/categories/products/categories/2"
pathname: "/_error"
route: "/_error"
...

```

It seems something wrong, but I really have no ideas.
I've tried this, but still same: https://github.com/vercel/next.js/discussions/22512
Any ideas?

---

### `Link` with href
```
<ul>
  {   mainNav.map(item => (
        <li>
          <Link passHref legacyBehavior href={"products/categories/1"}>
            <a target="_blank">{item.name}</a>
          </Link>
        </li>
      )
  )}
</ul>
```

### `Link` with href UrlObject
```
<ul>
  {   mainNav.map(item => (
        <li>
          <Link passHref legacyBehavior href={{
            pathname: "products/categories/[productType]",
            query: {
               productType: 1
            }
         }}>
            <a target="_blank">{item.name}</a>
          </Link>
        </li>
      )
  )}
</ul>
```

**Comment by icyJoseph** [comment, marked as answer]
You are missing the leading slash, try with, `/products/categories/1`:

```js
<ul>
  {   mainNav.map(item => (
        <li>
          <Link passHref legacyBehavior href={"/products/categories/1"}>
            <a target="_blank">{item.name}</a>
          </Link>
        </li>
      )
  )}
</ul>
```

**Reply by chen-chens** [reply]
@icyJoseph Oh my~~ I waste my whole day to find solution. I should notice that!
Really appreciate!!
